File: ios/Sources/CapacitorCalendarPlugin/Models/Results/CheckPermissionResult.swift
import EventKit
import Capacitor

struct CheckPermissionResult: JSResult {
    private let status: CAPPermissionState

    init(status: CAPPermissionState) {
        self.status = status
    }

    func toJSON() -> JSObject {
        var result = JSObject()
        result["result"] = status.rawValue
        return result
    }
}

File: android/src/main/java/dev/barooni/capacitor/calendar/models/results/CheckPermissionResult.kt
package dev.barooni.capacitor.calendar.models.results

import com.getcapacitor.JSObject
import com.getcapacitor.PermissionState
import dev.barooni.capacitor.calendar.models.templates.JSResult

data class CheckPermissionResult(
    val state: PermissionState,
) : JSResult {
    override fun toJSON(): JSObject {
        val result = JSObject()
        result.put("result", state.toString())
        return result
    }
}
